AsyncNats/NatsConnection.cs

```diff
@@ -157,32 +165,67 @@ private async Task ReadWriteAsync(CancellationToken disconnectToken)
 
                 try
                 {
-                    _logger?.LogTrace("Connecting to {Server}", Options.Server);
-                    await socket.ConnectAsync(Options.Server);
+                    _currentServer = _serverPool.SelectServer(_isRetry);
```

Comment:
See comment about IP being part of server selection.   I would move the IP resolving to the server-pool and let it handle selection from there.